src/forward_index.c

```diff
@@ -44,9 +44,12 @@ static uint32_t hashKey(const void *s, size_t n) {
 static size_t estimtateTermCount(const Document *doc) {
   size_t nChars = 0;
   for (size_t ii = 0; ii < doc->numFields; ++ii) {
-    size_t n;
-    RedisModule_StringPtrLen(doc->fields[ii].text, &n);
-    nChars += n;
+    DocumentField *field = doc->fields + ii;
+    if (field->unionType == FLD_VAR_T_CSTR || field->unionType == FLD_VAR_T_RMS) {
```

Comment:
What if it's no one of those types, don't we still need to estimate it? what this function is used for?

Reply:
`estimtateTermCount` is only relevant for stemming terms.

Reply:
So we will not reach here if it's not one of those values? If so let's assert instead of the if ...

Reply:
From what I can see, `estimtateTermCount` is used for stemming.

Reply:
@MeirShpilraien the assert fails if the index contains numeric, geo, or an array.

We use the result of the function, to set the size for the hash table of the forward index.

The new behavior is more correct for JSON, as it will result in a more balanced hash table for indexes with fields of other types, and does not change the behavior for HASH.

To fix this entirely we will need a link to the `FieldSpec` inside `DocumentField`. This might allow us to remove `name` and `path` from `DocumentField` and rely on those in `FieldSpec`. WDYT?

Reply:
Introduced flag `Index_FromLLAPI` since the LLAPI does not initiate `sp->rule`.